frontend/src/router/index.js

```diff
@@ -16,24 +16,34 @@ const router = createRouter({
     {
       path: '/dashboard',
       name: 'dashboard',
-      component: DashboardView
+      component: DashboardView,
+      meta:{ title: "Dashboard"}
     },
     {
       path: '/about',
       name: 'about',
-      component: AboutView
+      component: AboutView,
+      meta:{ title: "About"}
     },
     {
       path: '/faq',
       name: 'faq',
-      component: FaqView
+      component: FaqView,
+      meta:{ title: "FAQ"}
     },
     {
       path: '/contact',
       name: 'contact',
-      component: ContactView
+      component: ContactView,
+      meta:{ title: "Contact"}
     }
   ]
+  
+});
+
+router.beforeEach((toRoute, fromRoute, next) => {
+  window.document.title = toRoute.meta && toRoute.meta.title ? toRoute.meta.title : 'Home';
```

Comment:
@DavideDadomo un po' mi sono aiutato con chatgpt un po' lo sapevo già. 
è un po brodoso questo if inline e non mi fa impazzie.
Premessa, va benissimo eh però anche per te per darti qualche skill in più su javascript.

Esiste un modo piu carino e pulito per gestire queste cose ovvero:
`obj.data?.element`

cosa fa questo? senza fare l'if va a fare il check se esiste prima il data dell'oggetto e se esiste allora fa il .element, dunque quello che hai fatto potrebbe essere sostituito cosi:

`document.title = toRoute.meta?.title || 'Home';`

utilizziamo quelo che ti ho detto prima e la condiuzione OR che funge da default, ovvero 'Home'

per il primo pezzo di variabile invece mi sono fatto aiutare da chatgpt, non so ancora perchè ma usare window. non è un granche, dunque questo è migliore, credo sia piu versatile per cose future